inbox/models/backends/oauth.py

```diff
@@ -34,41 +36,65 @@ def log_token_usage(reason, refresh_token=None, access_token=None, account=None)
         access_hash=hash_token(access_token, prefix="access_token"),
         nylas_account_id=nylas_account_id,
         email=account.email_address if account else None,
+        scopes=scopes,
     )
 
 
 class TokenManager:
     def __init__(self):
-        self._tokens: Dict[str, str] = {}
+        self._token_cache: Dict[
+            Tuple[str, Optional[Tuple[str, ...]]], Tuple[str, datetime]
+        ] = {}
 
     def get_token(
         self,
         account: "OAuthAccount",
         force_refresh: bool = False,
         scopes: Optional[List[str]] = None,
     ) -> str:
-        if account.id in self._tokens:
-            token, expiration = self._tokens[account.id]
+        cache_key = self.get_cache_key(account, scopes)
```

Comment:
Those changes here are needed so we don't get tokens with wrong scopes from the cache. I was getting "Invalid audience" errors from Microsoft Graph and I realized this is because of this local in-process cache. Instead of caching on account ids we cache on pair (account_id, scopes) which prevents those errors.